stdlib/2/httplib.pyi

```diff
@@ -38,6 +39,15 @@ class HTTPResponse:
     def getheader(self, name, default: Optional[Any] = ...): ...
     def getheaders(self): ...
 
+# This is an API stub only for HTTPConnection and HTTPSConnection, as used in
+# urllib2.AbstractHTTPHandler.do_open, which takes either the class
+# HTTPConnection or the class HTTPSConnection, *not* an instance of either
+# class. do_open does not use all of the parameters of HTTPConnection.__init__
+# or HTTPSConnection.__init__, so HTTPConnectionProtocol only implements the
+# parameters that do_open does use.
+class HTTPConnectionProtocol(Protocol):
+    def __call__(self, host: str, timeout: int = ..., *, context: Optional[ssl.SSLContext] = ...) -> HTTPConnection: ...
```

Comment:
Good solution! But since `AbstractHTTPHandler.do_open()` just passes on any additional arguments, I feel it's best to allow any keyword arguments, not just context. Otherwise this can lead to false positives.

Reply:
Addressed in dc437c24